### PR TITLE
adding stats testcases

### PR DESCRIPTION
**Pull request by ANISH-GOTTAPU**
adding stats testcases for #123,#124,#140,#141

### CODE REVIEW

**Comment by ashutshkumr** [comment]
@ANISH-GOTTAPU please check if build failures are legitimate. If it's expected, please skip mark them as e2e unless the underlying issue is fixed.

**Comment by ANISH-GOTTAPU** [comment]
@Rangababu-R & @ashutshkumr Not able to get why this is failing in CI/CD setup, if port stats are matched then flow stats also should match properly. Can you pull and check in your setup once
<b>Local</b>
Fetching all port stats ...
Fetching all flow stats ...

Port Stats
-----------------------------------------------------------------------------------------------
           Port      Tx Frames       Tx Bytes      Rx Frames       Rx Bytes         Tx FPS
         raw_tx           3000        3074000              0              0            0.0
         raw_rx              0              0           3000        3074000            0.0
-----------------------------------------------------------------------------------------------


Flow Stats
--------------------------------------------------
           Flow      Rx Frames       Rx Bytes
             f1           1000          74000
             f2           2000        3000000
--------------------------------------------------


<b>CI/CD flow</b>

Fetching all port stats ...
Fetching all flow stats ...

Port Stats
-----------------------------------------------------------------------------------------------
           Port      Tx Frames       Tx Bytes      Rx Frames       Rx Bytes         Tx FPS
         raw_tx           3000        3074000              0              0            0.0
         raw_rx              0              0           3000        3074000            0.0
-----------------------------------------------------------------------------------------------


Flow Stats
--------------------------------------------------
           Flow      Rx Frames       Rx Bytes
             f2           2000        3000000
             f1              0              0
--------------------------------------------------

**Comment by ashutshkumr** [comment]
Maybe something wrong with calculation ? From logs I see this

```
        for row in flow_results:
            if row['name'] == 'f1':
                if column_name == 'frames_tx':
                    assert row[column_name] == f1_packets
                elif column_name == 'frames_rx':
>                   assert row[column_name] == f1_packets
E                   assert 0 == 1000
E                     -0
E                     +1000

```

**Comment by ANISH-GOTTAPU** [comment]
Yes the flow stats on f1 should be 1000, not sure why 0 is returned. f1(1000) & f2(2000) , total 3000 are showing in port stats but same is not reflecting in flow stats of f1. But same is working in local with python2 & python3

File: tests/statistics/test_port_and_flow_stats.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    """
+
+    f1_packets = 1000
+    f2_packets = 2000
+    f1_size = 74
+    f2_size = 1500
+    ports = b2b_raw_config.ports
+    flow1 = b2b_raw_config.flows[0]
+
+    flow1.packet = [
```
Comment (ashutshkumr) [inline tests/statistics/test_port_and_flow_stats.py:30]: no need to add header here

File: tests/statistics/test_port_and_flow_stats.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+    flow2.duration = flow.Duration(flow.FixedPackets(packets=f2_packets))
+    flow2.size = flow.Size(f2_size)
+    flow2.rate = flow.Rate(value=10, unit='line')
+
+    utils.start_traffic(api, b2b_raw_config, start_capture=False)
+    utils.wait_for(
+        lambda: utils.is_traffic_stopped(api), 'traffic to stop'
+    )
+
```
Comment (ashutshkumr) [inline tests/statistics/test_port_and_flow_stats.py:68]: Poll for stats to be as expected to signal that we have all required data available for validation. Do not use tolerance value.

File: tests/statistics/test_port_and_flow_stats.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                                             f2_packets,
+                                             f1_size,
+                                             f2_size):
+    """
+    Validate Port stats based on column_names
+    """
+
+    total_bytes = (f1_packets * f1_size) + (f2_packets * f2_size)
+    total_packets = f1_packets + f2_packets
+    tolerance = 10
```
Comment (ashutshkumr) [inline tests/statistics/test_port_and_flow_stats.py:131]: remove tolerance, and simplify assertions

File: tests/statistics/test_port_and_flow_stats_e2e.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    2) Get flow statistics based on flow name & column names and assert
+    each flow & column has returned the values and assert
+    """
+
+    f1_size = 74
+    f2_size = 1500
+    ports = b2b_raw_config.ports
+    flow1 = b2b_raw_config.flows[0]
+
+    flow1.packet = [
```
Comment (ashutshkumr) [inline tests/statistics/test_port_and_flow_stats_e2e.py:26]: remove headers

File: tests/statistics/test_port_and_flow_stats.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
+from abstract_open_traffic_generator import flow
+from abstract_open_traffic_generator import result
+import pytest
+
+
+@pytest.mark.e2e
```
Comment (ashutshkumr) [inline tests/statistics/test_port_and_flow_stats.py:6]: don't skip tests by tagging it as e2e. use something like:
```python
@pytest.mark.skip(reason="Reason for skip")
```

**Review by ashutshkumr — CHANGES_REQUESTED** [review]
Please address all review comments.

**Review by ashutshkumr — APPROVED** [review]

**Pull request merged**